go/vt/vttablet/onlineddl/executor.go

```diff
@@ -1826,17 +1826,16 @@ func (e *Executor) CancelMigration(ctx context.Context, uuid string, message str
 		// If this was not issued by the user, then this is an internal state machine cancellation of the
 		// migration, e.g. because it is stale or has an unrecoverable error. In this case we do not mark
 		// the timestamp, and as result, the state will transition to 'failed'
-		_ = e.updateMigrationTimestamp(ctx, "cancelled_timestamp", uuid)
+		if err := e.updateMigrationTimestamp(ctx, "cancelled_timestamp", uuid); err != nil {
+			return nil, err
+		}
 	}
 	defer e.failMigration(ctx, onlineDDL, errors.New(message))
 	defer e.triggerNextCheckInterval()
 
 	switch onlineDDL.Status {
 	case schema.OnlineDDLStatusQueued, schema.OnlineDDLStatusReady:
 		log.Infof("CancelMigration: cancelling %s with status: %v", uuid, onlineDDL.Status)
-		if err := e.updateMigrationStatus(ctx, onlineDDL.UUID, schema.OnlineDDLStatusCancelled); err != nil {
```

Comment:
these lines have been removed because call to `defer e.failMigration(ctx, onlineDDL, errors.New(message))` already takes care of updating the migration status.